### ISSUE TITLE
do not use `wgetopt_long_only()`

### ISSUE THREAD

**Issue by krader1961** [fish-shell/fish-shell#4114]
I decided to spend an hour making more code cleanups just because I wanted to do something that didn't require much thought. Specifically, normalizing more of the flag parsing by builtin commands to a consistent pattern. This time I was updating the `set` builtin flag  parsing. The code has a mixture of `wgetopt_long()` and `wgetopt_long_only()` calls. The `set` command was using the former. I happened to copy/paste an example using the latter. Perplexing test failures ensued.

Specifically, running `set -xl` reported an error. Which seems odd since both `-x` and `-l` are legal short flags for the `set` command. The problem is that `wgetopt_long_only()` treats `-xl` as equivalent to `-x --l` which is ambiguous because the command has long options `--long` and `--local`. This despite the fact it also has a `-l` short flag.

In short we do not want to ever use `wgetopt_long_only()`. So I'm going to change all current uses and add a warning to cppcheck to keep us from using it in the future.
